csrc/scheduler/normalization.cpp

```diff
@@ -170,9 +176,16 @@ std::shared_ptr<ReductionParams> innerOuterPersistentHeuristic(
   const int64_t workload_per_thread = inner_dim_numel >= 4096 ? 4 : 2;
   iop.vectorization_factor_outer =
       std::min(workload_per_thread, max_tmp_gmem_vect_factor);
-  iop.bdimx = scheduler_utils::roundUpPow2(
+  // threads_per_block has factor of 8, roundup to increase the probability of
+  // bdimx * bdimy == threads_per_block.
+  iop.bdimx = scheduler_utils::roundUpPow2Or8(
       ceilDiv(inner_dim_numel / iop.vectorization_factor_outer, iop.gdimy));
-
+  // if still not dividable, e.g. threads_per_block = 256, bdimx = 40.
+  // increase bdimx to make it dividable. This will avoid the increase of
+  // threads per block which may cause register spills.
+  while (threads_per_block % iop.bdimx) {
+    iop.bdimx = std::min(iop.bdimx + 8, threads_per_block);
+  }
   // Step-4, set OuterParams Reduction dim: bdimy.
   iop.bdimy = ceilDiv(threads_per_block, iop.bdimx);
```

Comment:
If they are still not divisible, it seems `iop.bdimy * iop.bdimx` would be larger than `threads_per_block`. Isn't it a concern?

Reply:
This won't happen because `iop.bdimx` will be set to `threads_per_block` in the while loop and then `iop.bdimy` is 1.

Reply:
They, why is `ceilDiv` used? Using `ceilDiv` seems to indicate they may not always divisible. If that's not the case, asserting they are divisible and just using `/` would make the code much clearer.

Reply:
changed to:
```
  iop.bdimy = threads_per_block / iop.bdimx;
  TORCH_INTERNAL_ASSERT(
      iop.bdimy * iop.bdimx == threads_per_block,
      " threads_per_block must be divisible by bdimx and bdimy.");
```